tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,235 @@
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    toolName="$1"
+    eval_tools
+    osName="$(uname -s)"
+
+    if $(echo "$osName" | grep -iqF "Darwin"); then
+        toolsSearchPath="$repoRoot/$SearchPathOSXTools"
+    else
+        toolsSearchPath="$repoRoot/$SearchPathLinuxTools"
+    fi
+
+    if [ -z "$toolsSearchPath" ]; then
+        echo "Unable to read tool search path for $toolName"
+        exit 1
+    fi
+
+    echo "$toolsSearchPath"
+}
+
+# Compares the version of the tool at the specified path with the declared version of the tool.
+# Each tool has to implement its own is_declared_version.sh script that performs version comparison.
+# This function invokes is_declared_version.sh corresponding to the tool.
+# Exit 1 if -
+#   1. The tool does not exist at the given path
+#   2. is_declared_version.sh script corresponding to the tool is not found
+#   3. The version of the tool at the given path does match the declared version.
+is_declared_version()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool path is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    if [ ! -f "$2" ]; then
+        echo "Tool path does not exist or is not accessible. Path: $2"
+        exit 1
+    fi
+
+    toolName="$1"
+    lowercaseToolName="$(echo $toolName | awk '{print tolower($0)}')"
+    toolPath="$2"
+    shellScriptsPath="$(get_repository_shell_scripts_path)"
+    declaredVersion=$(get_declared_version "$toolName")
+
+    if [ $? -eq 1 ]; then
+        echo "$declaredVersion"
+        exit 1
+    fi
+
+    overriddenIsDeclaredVersion="$shellScriptsRoot/$lowercaseToolName/is_declared_version.sh"
+
+    if [[ ! -z "$overriddenIsDeclaredVersion" && -f "$overriddenIsDeclaredVersion" ]]; then
+        $("$overriddenIsDeclaredVersion" "$toolPath" "$declaredVersion")
+
+        if [ $? -eq 1 ]; then
+            exit 1
+        fi
+    else
+        echo "Unable to locate is_declared_version.sh at the specified path. Path: $overriddenIsDeclaredVersion"
+        exit 1
+    fi
+}
+
+# Gets the error message to be displayed when the specified tool is not available for the build.
+# Each tool has to implement its own tool_not_found_message.sh script that returns the error message specific to the tool.
+# This function invokes tool_not_found_message.sh corresponding to the tool.
+# Exit 1 if the tool_not_found_message.sh is not found.
+tool_not_found_message()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    toolName="$1"
+    lowercaseToolName="$(echo $toolName | awk '{print tolower($0)}')"
+    shellScriptsPath="$(get_repository_shell_scripts_path)"
+    declaredVersion=$(get_declared_version "CMake")
```

Comment:
Fixed.

---
In reply to: [107794826](https://github.com/ravimeda/corefx/pull/2#discussion_r107794826) [](ancestors = 107794826)